### ISSUE TITLE
redirection handles Unicode URL changes incorrectly

### ISSUE THREAD

**Issue by gqqnbig** [johngodley/redirection#860]
### Steps to reproduce

I have an article whose URL contains Unicode charachters, eg. https://gqqnbig.me/2018/03/20/论markdown/

![chrome_2018-03-25_09-45-31](https://user-images.githubusercontent.com/614159/37877521-72ccf0ae-3011-11e8-8076-2eedb88d9795.png)

I now change the publish date to 2018-03-**21**, the URL is thus changed to https://gqqnbig.me/2018/03/21/论markdown/

Redirection captures the URL change, but the unicode character is encoded.

![chrome_2018-03-25_09-48-15](https://user-images.githubusercontent.com/614159/37877547-bb2d687e-3011-11e8-884e-9728c39383d1.png)

When I visit the old URL, ie. https://gqqnbig.me/2018/03/20/论markdown/ or https://gqqnbig.me/2018/03/20/%e8%ae%bamarkdown/ , the page is 404.

If I change Redirections rules to be the actual Unicode characters, when I visit the old URL, I'm successfully redirected to the new page.

![chrome_2018-03-25_09-50-57](https://user-images.githubusercontent.com/614159/37877568-08c0cea0-3012-11e8-9d2e-b0306dc37557.png)

### Environment

WordPress 4.9.4 Chinese version

Redirection 3.2

The server that runs WordPress is Windows Server 2012 English version.

**Comment by johngodley** [comment]
Fixed in [4.3 alpha 1](https://github.com/johngodley/redirection/releases/tag/4.3-alpha-1), which will be released at some point in the future. Please let me know how it goes.

Closing this issue as complete.

**Comment by gqqnbig** [comment]
Hi @johngodley 

I installed 4.3 alpha 1, but it crashes my website. 

```
[Fri Apr 26 06:57:40.371975 2019] [:error] [pid 23339] [client 54.36.149.220:32280] 

PHP Fatal error:  Uncaught Error: Call to undefined function mb_strtolower() in /var/www/html/wp-content/plugins/redirection/models/url-match.php:46
Stack trace:
#0 /var/www/html/wp-content/plugins/redirection/models/redirect.php(117): Red_Url_Match->get_url()
#1 /var/www/html/wp-content/plugins/redirection/models/redirect.php(107): Red_Item::get_for_matched_url(Object(Red_Url_Match))
#2 /var/www/html/wp-content/plugins/redirection/modules/wordpress.php(113): Red_Item::get_for_url('/2019/04/25/\\xE7\\xBE\\xA4...')
#3 /var/www/html/wp-includes/class-wp-hook.php(286): WordPress_Module->init('')
#4 /var/www/html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
#5 /var/www/html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#6 /var/www/html/wp-settings.php(505): do_action('init')
#7 /var/www/html/wp-config.php(93): require_once('/var/www/html/w...')
#8 /var/www/html/wp-load.php(37): require_once('/var/www/html/w...')
#9 /var/www/html/wp-blog-header.php(13): require_once('/var/www/html/w...')
#10 /var/ww in /var/www/html/wp-content/plugins/redirection/models/url-match.php on line 46
```

I don't really know PHP, but a search shows mb_strtolower is available on PHP 4 >= 4.3.0, PHP 5, PHP 7. From the screenshot, I can see my php version should be 7.0.

![chrome_2019-04-26_00-17-29](https://user-images.githubusercontent.com/614159/56790196-c6e43a00-67b8-11e9-8206-5cfde33166cb.png)

If you need other information, let me know.

**Comment by johngodley** [comment]
Interesting. Yep, you don't have that function enabled on your PHP. I'll be releasing a beta version soon so I'll make sure to handle this situation.

**Comment by johngodley** [comment]
This should be fixed in the 4.3 beta:

https://github.com/johngodley/redirection/releases/tag/4.3-beta-1

**Comment by gqqnbig** [comment]
4.3 beta works for me. Thanks!
